python/cudf/cudf/core/reshape.py

```diff
@@ -190,7 +190,26 @@ def concat(objs, axis=0, ignore_index=False, sort=None):
     if len(objs) == 0:
         raise ValueError("All objects passed were None")
 
-    typs = set(type(o) for o in objs)
+    # Retrieve the base types of `objs`. In order to support sub-types
+    # and object wrappers, we use `isinstance()` instead of comparing
+    # types directly
+    typs = set()
+    for o in objs:
+        if isinstance(o, DataFrame):
+            typs.add(DataFrame)
+        elif isinstance(o, Series):
+            typs.add(Series)
+        elif isinstance(o, cudf.MultiIndex):
+            typs.add(cudf.MultiIndex)
+        elif issubclass(type(o), Index):
+            typs.add(Index)
```

Comment:
All these classes are subclasses of class `cudf.core.frame.Frame`. Can we just use that?

Reply:
As far as I can see, the types are used to dispatch the concatenation to the right implementation: https://github.com/rapidsai/cudf/blob/70182b231be6c3a0b0b238cac9c2534dd22d4f26/python/cudf/cudf/core/reshape.py#L256-L281